tests/serverpod_test_server/test_integration/test_tools/database_operations_in_endpoint_test.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:test/test.dart';
+
+import 'serverpod_test_tools.dart';
+
+void main() {
+  withServerpod(
+    (endpoints, session) {
+      group('Given TestToolsEndpoint', () {
+        test(
+            'when calling createSimpleData then creates a SimpleData in the database',
+            () async {
+          await endpoints.testTools.createSimpleData(session, 123);
+
+          final result = await SimpleData.db.find(session);
+          expect(result.length, 1);
+          expect(result.first.num, 123);
```

Comment:
Suggestion: Add an internal copyWith on the session inside the test method call. This is to force a new session id and a new state for things such as the logger to mimic the production behaviour and not trick people into thinking the session is persistent between calls.

Reply:
[a84e1c5](https://github.com/serverpod/serverpod/pull/2713/commits/a84e1c5bf51d08574f40bf118f5e942011e97729)